Release 2.11.0 (#5684)
timescale/timescaledb: CHANGELOG.md

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

> ## 2.11.0 (2023-05-12)

Reviewer: Don't forget to update the date once we finalize the release date

This release contains new features and bug fixes since the 2.10.3 release.
We deem it moderate priority for upgrading.

This release includes these noteworthy features:

> * compressed hypertable enhancements:

Reviewer 2: ```suggestion
* Compressed hypertable enhancements:
```

> * Compressed hypertable enhancements:
>   * UPDATE/DELETE support
>   * ON CONFLICT DO UPDATE

Reviewer: ```suggestion
* Support for DML operations on compressed chunks:
  * UPDATE/DELETE support
  * Support for unique constraints on compressed chunks
  * Support for `ON CONFLICT DO UPDATE`
  * Support for `ON CONFLICT DO NOTHING`
```

> * Join support for hierarchical Continougs Aggregates

Reviewer 2: ```suggestion
* Join support for hierarchical Continuous Aggregates
```

> * Join support for hierarchical Continuous Aggregates

Reviewer 3: ```suggestion
* Join support for Hierarchical Continuous Aggregates
```

> * performance improvements

Reviewer 2: ```suggestion
* Performance improvements
```

> * Performance improvements

Reviewer: Removing this line cause `performance improvements` is too generic - if we have a notable performance improvement to mention, we should do so explicitly (e.g. `compression is 3123 times faster` or `queries over compressed chunks that filtrer by the segment by column are 2897 times faster`)

```suggestion
```

> ## Unreleased

Reviewer 4: We can remove the "unreleased" header.

Author: ok; and add it back in the `post-release` stuff

## 2.11.0 (2023-05-12)

> 

Reviewer 4: we generally add this upgrade recommendation based on severity of the fixes, something like.

"This release contains bug fixes since the 2.10.3 release.
We recommend that you upgrade at the next available opportunity."

Author: as it also contains features; I've added:
```
This release contains new features and bug fixes since the 2.10.3 release.
We deem it moderate priority for upgrading.
```
